components/style/properties/shorthand/border.mako.rs

```diff
@@ -131,14 +131,38 @@ pub fn parse_border(context: &ParserContext, input: &mut Parser)
 
     impl<'a> LonghandsToSerialize<'a>  {
         fn to_css_declared<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            let all_equal = {
+                % for side in PHYSICAL_SIDES:
+                  let border_${side}_width = self.border_${side}_width;
+                  let border_${side}_style = self.border_${side}_style;
+                  let border_${side}_color = self.border_${side}_color;
+                % endfor
+
+                border_top_width == border_right_width &&
+                border_right_width == border_bottom_width &&
+                border_bottom_width == border_left_width &&
+
+                border_top_style == border_right_style &&
+                border_right_style == border_bottom_style &&
+                border_bottom_style == border_left_style &&
+
+                border_top_color == border_right_color &&
+                border_right_color == border_bottom_color &&
+                border_bottom_color == border_left_color
+            };
+
             // If all longhands are all present, then all sides should be the same,
             // so we can just one set of color/style/width
```

Comment:
If this happens not to build, likely because of the unit tests, please remove this comment.

If you're busy or anything and can't rebase (these are minor changes), please feel free to mark the "Allow edits from maintainers" checkbox, and I'll land it for you with the changes (maintaining authorship of course).

Sorry for the extreme delay reviewing :(